assemblyline/odm/models/ontology/results/network.py

```diff
@@ -57,7 +57,24 @@ class NetworkConnection(odm.Model):
     connection_type = odm.Optional(odm.Enum(values=['http', 'dns', 'tls'], description="Type of connection being made"))
 
     def get_oid(data: dict):
-        return f"network_{get_dict_fingerprint_hash({key: data.get(key) for key in OID_PARTS})}"
+        connection_type = data.get('connection_type')
+        hash_dict = {key: data.get(key) for key in OID_PARTS}
+        oid_prefix = "network"
+        if connection_type == "http":
+            oid_prefix = "network_http"
+            http_details = data.get('http_details', {})
+
+            # Include any details involved in the request for hashing
+            hash_dict['http_details'] = {
+                field: http_details.get(field)
+                for field in NetworkHTTP.fields().keys() if field.startswith('request_')
+            }
+        elif connection_type == "dns":
+            # Include the requested domain as part of the hash
+            oid_prefix = "network_dns"
+            hash_dict['dns_details'] = {'domain': data.get('dns_details', {}).get('domain', None)}
```

Comment:
If you wanted to have all "query" information and no "response" information, I believe the `lookup_type` should also be part of the dns_details.